modules/tensor_mechanics/include/materials/StrainEnergyDensity.h

```diff
@@ -0,0 +1,45 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef STRAINENERGYDENSITY_H
+#define STRAINENERGYDENSITY_H
+
+#include "Material.h"
+#include "DerivativeMaterialInterface.h"
+
+class RankTwoTensor;
+
+class StrainEnergyDensity;
+
+template <>
+InputParameters validParams<StrainEnergyDensity>();
+
+/**
+ * StrainEnergyDensity calculates the strain energy density.
+ */
+class StrainEnergyDensity : public DerivativeMaterialInterface<Material>
+{
+public:
+  StrainEnergyDensity(const InputParameters & parameters);
+
+  virtual void initQpStatefulProperties() override;
+  virtual void computeQpProperties() override;
+
+protected:
```

Comment:
Provide doxygen comments for all of the member variables below.  You don't need them for any of the methods because they're all overrides.